CIPs/cip-107.md

```diff
@@ -8,38 +8,135 @@ Created: 2023-02-06
+* `used_storage_points`: storage points already used for storage collateral. Accessible via `cfx_getSponsorInfo` RPC or `SponsorWhitelistControl` internal contract.
+* `tx.value`: value field of the transaction, meaning the amount of CFX transferred with the transaction.
+
+#### Composition
+
+##### Before CIP-107
+
+Prior to CIP-107, the CFX provided by a collateral sponsor would be completely refunded if the sponsor was replaced or the contract was destroyed:
+
+`CFX provided by collateral sponsor = refundable collateral`
+
+This refundable collateral, however, cannot be directly accessed on-chain. It can be computed from two components: the contract's `collateral_for_storage` and `sponsor_balance_for_collateral` (accessible via `cfx_getSponsorInfo` RPC or `SponsorWhitelistControl` contract):
+
+`CFX provided by collateral sponsor (i.e., refundable collateral) = collateral_for_storage + sponsor_balance_for_collateral`
+
+##### After CIP-107
+
+Following CIP-107's activation, the CFX provided by a collateral sponsor will not be fully refundable if the sponsor is replaced or the contract is destroyed. This is because a proportion of CFX is burned and converted into storage points (1024 storage points/CFX):
+
+`CFX provided by collateral sponsor = refundable collateral + (storage points / 1024)`
+
+Both parts above cannot be directly accessed on-chain. However, several new interfaces become available via `cfx_getSponsorInfo` RPC or `SponsorWhitelistControl` contract after CIP-107: `available_storage_points` and `used_storage_points`. The former represents available storage points for collateral, while the latter represents already used storage points:
+
+`storage points = available_storage_points + used_storage_points`
+
+The contract's `collateral_for_storage` now consists of both `used_storage_points` and the used refundable part:
+
+`CFX provided by collateral sponsor = collateral_for_storage + (sponsor_balance_for_collateral + available_storage_points / 1024)`
+
+and,
+
+`refundable collateral = sponsor_balance_for_collateral + (collateral_for_storage - used_storage_points / 1024)`
+
+#### Storage Points Conversion
+
+##### Sponsor Initiation
+
+When a sponsor initiates the sponsor balance for collateral, *p* proportion of the provided CFX is burned and converted into storage points:
+
+`CFX burnt = (storage points) / 1024 = p * (tx.value)`
+
+##### Lazy Conversion for Sponsored Collateral Before CIP-107 Activation
+
+Any CFX provided as sponsor balance for collateral before CIP-107 activation will also be burned based on *p*:
+
+`CFX burnt = (storage points) / 1024 = p * (CFX ever provided) = p * (collateral_for_storage + sponsor_balance_for_collateral)`
+
+Note: the conversion is carried out lazily, and the conversion ratio may change following a DAO vote. The conversion will be triggered once the storage of the contract is changed at the transaction granularity.
```

Comment:
`will be triggered once the storage of the contract is changed` to `will be triggered once the storage collateral of the contract is changed`